Improve action stories

uthbees opened the pull request:
I finally got annoyed enough with the stories that I decided to put some effort into improving them again. This fixes all the typos and grammar issues (in action stories) that my IDE could find, and copies most of the new stories our group wrote into the file, because apparently no one ever did that. Credit for these stories goes to Inferno Vulpix, Nikki, and me (known as scout on the discord).

In regards to Vera's stories, I think they're generally great, and some are really amazing! They're not always as good as they could be, though, so when there was a story by Vera in the same slot as a story from the spreadsheet, I tried to use whichever one made the stories better overall.

There are still a few more things that I'd like to change, but I don't know when (or if) I'll get around to doing them, so it's probably best not to wait for that. I can always make another PR.

If there's anything in this PR that you want changed, let me know (by replying to this PR - I probably won't notice if you try to DM me on discord). Or I guess you could do it yourself - either way.

uthbees committed:
Fix grammar and typos

uthbees committed:
Copy over the already-written stories

Credit to Inferno Vulpix, Nikki, and scout (me) for the stories.

I think Vera's stories are generally great, and some are amazing! They're not always perfect, though, so when there was a story by Vera in the same slot as a story from the spreadsheet, I tried to use whichever one made the stories better overall.

TODOs:
- I left all the ruins and haul actions alone for now, since we only have (non-placeholder) stories written for a few of those.
- Some of the training action stories could use further improvement.
- I love IV's Wizard College stories, but Vera's are good too. Need to figure out what to do about that.
- IV wrote a cool set of stories for Fell From Grace with more and more required resources.  The numbers probably need some tweaking.

uthbees committed:
Merge branch 'dmchurch:dmchurch' into dmchurch

dmchurch commented:
Oh my goodness, this is _fantastic!_ I'm not merging it in right this moment, but that's only because I want to take a little bit of care with adding new action stories, given the fragility of the current system - I know you saw some of the issues there, around renumbering/reordering, and I also want to make sure that the Action Log (which records story completions by index-in-list as well, since that was the only thing available to me at the time) doesn't get confused. I'm likely to take this as an opportunity to put _some_ sort of non-position-based key onto action stories, so they can get moved around at need.

Oh, and thanks for the whitespace fixes, I appreciate them - I'm not as diligent about that as perhaps would be helpful 😂

uthbees commented:
Sounds great! I did test everything and it all works as far as I can tell, with the only weirdness being the new stories showing up in the log when you load the changes for the first time and (I think) that the two existing judgement stories show up in the log again if you already have them unlocked because of my reordering hack. And when I changed story requirements, I believe I only ever decreased them, so no one should have anything unlocked that they shouldn't either. (Not that you shouldn't check yourself - it's entirely possible that I missed something.)

uthbees committed:
Second pass on stories

Looking at it with fresh eyes, there were a few more stories from IV that I thought would be worth putting in. These changes are less clear-cut, but I think they're still all good.

I also made a few miscellaneous edits, most notably making the difference between the Magic and Restoration skills more clear.

uthbees committed:
Put in IV's wizard college stories

Putting this as a separate commit because it's more arguable than all the other changes. It even changes the balance of the game (very, very slightly) by reducing the number of wizard college ranks by one.

uthbees committed:
Render stories in the order they appear in game.xml

uthbees committed:
Remove hacks

There are still problems if a story is changed to something that unlocks later than the original, but there's no way of getting around that. Even if we added new ways of identifying stories, we still wouldn't be able to reuse identifications in those situations.

uthbees commented:
I did a second pass on the spreadsheet and moved over a few more stories.

Also, it looks like my empty stories do show up in the action log - I must have just missed them earlier because of the log only updating at certain times. I went ahead and took a stab at a fix by changing the relevant functions to render the stories in the order they appear in game.xml instead of in numerical order. (This makes my hacks unnecessary, so I took them out.)

What do you think of this solution? I'm not really the one who would have to live with it, after all...

dmchurch commented:
Oh wow you're willing to do the work! In that case I'm gonna go through this with a fine-combed tooth and give you some notes for stuff you can continue on in this PR 😉 (and feel free to decline any of these - this is stuff that I've already planned on doing "at some point", so it's not stuff you _have_ to do!)

dmchurch commented on actionList.js:
```diff
@@ -5081,17 +5088,14 @@ Action.WizardCollege = new MultipartAction("Wizard College", {
+        if (curWizCollegeSegment >= 30) unlockStory("wizardGuildSageAchieved");
+        if (curWizCollegeSegment >= 42) unlockStory("wizardGuildMagusAchieved");
+        if (curWizCollegeSegment >= 54) unlockStory("wizardGuildCouncilMemberAchieved");
+        if (curWizCollegeSegment >= 57) unlockStory("wizardGuildChairAchieved");
```
Let's simplify this all a bit. We've got a number of story flags which are "Did X of a thing", "Did Y of a thing", and it's all so much boilerplate. Here's my thought:

1. change `storyReqs` to `storyFlags` to make its purpose clearer. Your call on whether to do this as a global refactor or just by changing the name of the variable and adding a `const storyReqs = storyFlags` to avoid breaking code. this will also bring it into line with the content editor I'm writing, which calls them "Story flags". If you do make this a global refactor, be sure to search all instances of the string "storyReqs" and don't just rely on built-in reference finding, which (for example) won't find references from html files. **However, the key in the save format should still be `"storyReqs"`, for export compatibility.**
2. rename `unlockStory()` to `setStoryFlag()`; in this case I'd rather you not do it as a full refactor, just add a `const unlockStory = setStoryFlag` with a note to refactor it later.
3. add a parallel `storyVars` structure; this one should be a `Record<string, number>` rather than `Record<string, boolean>`, and it should include `maxWizardGuildSegmentCleared: -1` in its initializer. (oh and, feel free to add a `@satisfies` clause to both storyFlags and storyVars.) It will need the appropriate load/save logic, and also to be registered to the Data registry like storyFlags.
4. add a new function `increaseStoryVarTo(name, value)` which sets the named story variable to the max of its current level and `value`, and triggers the action log logic if it increased.
5. add a new structure to saving.js for defining migration of story flags/vars from old names/types to new. It should have a form like the following:
   ```ts
   const storyInitializers: {
     storyFlags: {
       [K in keyof typeof storyFlags]?: (loadingFlags: Record<string, boolean>, loadingVars: Record<string, number>) => boolean
     },
     storyVars: {
       [K in keyof typeof storyVars]?: (loadingFlags: Record<string, boolean>, loadingVars: Record<string, number>) => number
     },
   }
   ```
   and add the appropriate loading logic to call the relevant method when loading from a save that doesn't have a given flag/var set. Don't worry about trying to maintain export compatibility here.
6. Change this code to the obvious.

dmchurch commented on actionList.js:
```diff
@@ -5106,33 +5110,31 @@ Action.WizardCollege = new MultipartAction("Wizard College", {
     finish() {
-        //guild = "Wizard";
         resources.wizardCollege = true;
         unlockStory("wizardGuildTestTaken");
```
Becomes `increaseStoryVarTo("maxWizardGuildSegmentCleared", 0)`

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2585,6 +2592,8 @@
             <!--Referencing Roy Mustang from Fullmetal Alchemist.-->
             <story_4><![CDATA[<b>1,000 Pyromancy:</b>⮀ You snap your fingers idly a few times, getting a feel for the motion before turning your attention to a piece of wood you prepared. With one last snap, the entire length of the wood catches fire all at once, and within moments a cozy campfire crackles merrily before you. Mundane as such a trick may seem, making fire just from the friction be ...
+            <!-- Alternate 1,000 pyromancy story from IV:
+               The fire is conjured and extinguished in tune with your breath, an extension of your will as much as your arms and legs. In a fight you are the burning battlefield itself, every wisp of flame bends to your command and seeks only one thing: the complete incineration of your foes in a ravenous inferno. And then the battle ends and you withdraw the fire back into yourself, where it wa ...
```
Feel free to add this as a story at 500 pyro if you like!

uthbees replied:
Hmm, it doesn't feel right to me to have a slow-paced story after a really dramatic story like that. If we're including IV's story here, I think it should be the last one in the series - what do you think of putting the Fullmetal Alchemist reference at 500? We'd just need to replace the last line with something less strong - maybe something like "Pathetic."?

Either way is fine - I don't know how attached you are to the current 1k story.

dmchurch replied:
a bit attached - I wrote it! (or, well, rewrote it to improve the FMA reference.) I was thinking about suggesting moving that to 500, but I'm still torn. I sort of like the idea of "anyone can make a big fire, but mastery is fine control" or that sort of thing, but the problem is that you're _right_, the pacing is off, and honestly I was thinking the same.

what would you think of adding a closing sting to the 1k, something that implies that it comes at the tail end of a lot of violence, like "behind you, the charred remains of the mountainside continue to smolder"? I'll leave it up to you: either amend the 1k like that to improve the pacing, move it to 500 and edit it so that you're matching rather than exceeding the ancients (and you could add a callback to Mustang in the 1k, if you wanted, "even Mustang couldn't have imagined" or similar), or just leave it alone and we'll come back to it later 😄

uthbees replied:
I tried moving the 1k to 500, but the pacing was still off. It would be fine on its own, but next to the other pyromancy stories, I found the difference to be jarring. I ended up having to basically rewrite the story to bring the length down enough to match the others. I like how it came out, but it's kind of not what you wrote anymore (although I did get the smoldering mountainside thing in, that was nice).

If you want to keep your story in as it was before, I can revert the commit (it is your fork, after all). What do you think?

dmchurch replied:
I like it a lot! The most important thing is that my girlfriend (who introduced me to FMA) got to see the full story during her playthrough 😂 Also, the pacing works great - I like how now, 500 Pyro is where "you can do as well as the ancients if you've made yourself the tools" and 1k is just "you are the literal embodiment of fire, tools what tools"

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2612,7 +2621,7 @@
 			<story_1><![CDATA[<b>Shiny spot checked:</b>⮀ You chip away at the wall until the sparkling object that caught your eye falls free. To your disappointment, what comes out of the wall is just some shiny rock. You doubt it'd even be worth much at market.]]></story_1>
 			<story_2><![CDATA[<b>Soulstone mined:</b>⮀ This time you're sure it's a soulstone, and as you pry it out of the wall you feel a rush of vindication. It glows an earthy brown with a slight blue tint, in contrast to the colors you found in the dungeon.]]></story_2>
-			<story_3><![CDATA[<b>30 Soulstones mined:</b>⮀ As you found all the soulstones hiding in Mt. Olympus' cavern walls, you quickly realized that unlike the soulstones found in dungeons, these soulstones appear in every loop. You wonder what the cause of that might be, but whatever the reason it's pretty convenient.]]></story_3>
+			<story_3><![CDATA[<b>30 Soulstones mined:</b>⮀ As you found all the soulstones hiding in Mt. Olympus' cavern walls, you quickly realize that unlike the soulstones found in dungeons, these soulstones appear in every loop. You wonder what the cause of that might be, but whatever the reason it's pretty convenient.]]></story_3>
```
```suggestion
			<story_3><![CDATA[<b>30 Soulstones mined:</b>⮀ As you found all the soulstones hiding in Mt. Olympus' cavern walls, you quickly realize that unlike the soulstones found in dungeons, these soulstones appear in every loop. You wonder what the cause of that might be; but, whatever the reason, it's pretty convenient.]]></story_3>
```

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2693,18 +2702,19 @@
-			<story_3><![CDATA[<b>250 Imbuement completed:</b>⮀ You're smarter, faster, stronger and wiser at the start of each loop, making the trek from Beginnersville to the mountain that much easier. Sure, you know that you have to give up some of your talent, but starting out better is well worth the price of admission.]]></story_3>
-			<story_4><![CDATA[<b>500 Imbuement completed:</b>⮀ As you step out of the pool, you sigh a little. You know you could imbue yourself further, but your mind wouldn't handle the strain. So, since you're out of ways to further improve your mind, you confirm that you're basically done with the glowing pool.]]></story_4>
+			<story_1><![CDATA[<b>Third phase reached:</b>⮀ Even if the silvery glow of the lake has faded, this is still a place of power. You've been able to reinforce your mind in a way that persists through the loops, so why shouldn't you be able to do the same with your body? You meditate in the clear waters, trying to find the core of your being.  There are no soulstones arrayed around you: you don't ...
+			<story_2><![CDATA[<b>Imbuement complete:</b>⮀ Your breath stills, the very blood in your veins crawls to a halt for the briefest of moments, as you finally reach your innermost self. With nowhere to go but forward, you press on, trying to fill the space within with what you've gathered in your practice and training. An indescribable feeling washes over you as a tiny sliver of your strength fad ...
```
```suggestion
			<story_2><![CDATA[<b>Imbuement complete:</b>⮀ Your breath stills; the very blood in your veins crawls to a halt for the briefest of moments, as you finally reach your innermost self. With nowhere to go but forward, you press on, trying to fill the space within with what you've gathered in your practice and training. An indescribable feeling washes over you as a tiny sliver of your strength fades, invested into your future self. The needs of your body reassert themselves the moment you're done, and you're left gasping for air in the middle of a subterranean cave.]]></story_2>
```

uthbees replied:
Fixed the weird sentence by changing the "as" to an "and." (The version with the semicolon is still weird to me.)

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2693,18 +2702,19 @@
-			<story_4><![CDATA[<b>500 Imbuement completed:</b>⮀ As you step out of the pool, you sigh a little. You know you could imbue yourself further, but your mind wouldn't handle the strain. So, since you're out of ways to further improve your mind, you confirm that you're basically done with the glowing pool.]]></story_4>
+			<story_1><![CDATA[<b>Third phase reached:</b>⮀ Even if the silvery glow of the lake has faded, this is still a place of power. You've been able to reinforce your mind in a way that persists through the loops, so why shouldn't you be able to do the same with your body? You meditate in the clear waters, trying to find the core of your being.  There are no soulstones arrayed around you: you don't ...
+			<story_2><![CDATA[<b>Imbuement complete:</b>⮀ Your breath stills, the very blood in your veins crawls to a halt for the briefest of moments, as you finally reach your innermost self. With nowhere to go but forward, you press on, trying to fill the space within with what you've gathered in your practice and training. An indescribable feeling washes over you as a tiny sliver of your strength fad ...
+			<story_3><![CDATA[<b>50 Imbuement completed:</b>⮀ Each time you come back here, it's with the intent to leave as something less than when you entered, the capability you've honed over many loops sacrificed to fill the depths of your being with a power that transcends time. No more do you begin each loop as the hapless courier who couldn't even keep a precious package safe; with every visit tha ...
```
```suggestion
			<story_3><![CDATA[<b>50 Imbuement completed:</b>⮀ Each time you come back here, it's with the intent to leave as something less than when you entered, the capability you've honed over many loops sacrificed to fill the depths of your being with a power that transcends time. No more do you begin each loop as the hapless courier who couldn't even keep a single precious package safe; with every visit that person becomes more and more of a distant memory. Every time you return to that moment in Beginnersville, the resumed flow of time sees you moving faster, thinking clearer, and fighting better than your past self ever did, and you're glad for it. What better joy is there, you ask yourself, than self-improvement?]]></story_3>
```

uthbees replied:
Adding "single" here would imply that the character lost _all_ precious packages to me. I definitely like the other suggestion though, you're right that the loop doesn't start with waking up. Maybe IV was thinking of MoL.

dmchurch replied:
Yeah, I waffled hard on whether or not to add "single" for that very reason 😂 I think the new version looks great!

dmchurch commented on actionList.js:
```diff
@@ -5033,17 +5042,15 @@ Action.WizardCollege = new MultipartAction("Wizard College", {
+            case 17: return storyReqs.wizardGuildSageAchieved;
+            case 18: return storyReqs.wizardGuildMagusAchieved;
+            case 19: return storyReqs.wizardGuildCouncilMemberAchieved;
+            case 20: return storyReqs.wizardGuildChairAchieved;
```
Most of these already have an associated index; now that we have the ability to use out-of-order indices (thank you for that!) go ahead and return these to their old indices and insert the new stories in the right place with out-of-order numbers.

dmchurch commented on actionList.js:
```diff
@@ -5806,8 +5808,9 @@ Action.RaiseZombie = new Action("Raise Zombie", {
-                case 2: return storyReqs.failedRaiseZombie;
+            case 1: return storyReqs.attemptedRaiseZombie;
+            case 3: return storyReqs.raised10Zombies;
+            case 4: return storyReqs.raised25Zombies;
```
These can become storyVars, like WC

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -930,40 +930,46 @@
-                house-to-house charity collections, there are museums and shops where you can find wonders beyond the reaches of mere mortals,
-                and you even learn there are opportunities to become a registered citizen, even as a mere mortal! Whether you want to study at the
-                Wizard's college or fight alongside the Valkyries, there are many benefits to living among the Gods.
+                You expected to be whisked away to a land of opulence and wonder beyond your wildest dreams. Instead,
+                you sat in a cramped room filling out paperwork for three hours before being dropped off outside what
+                appears to be an ordinary town, with manicured lawns and meandering sidewalks leading to tidy parks and
+                decorations. It's not ordinary though, if you stare at the grass you notice it's just a bit too green,
+                the sky just a bit too blue. The literal deity taking a seven foot tall dog on a walk doesn't help
+                either. A god passing by asks if you want to contribute to a local charity, and you see another god
+                offering tours of 'Valhalla, the greatest city above the world!'
```
Mmh. Chefkiss, lovely.

uthbees replied:
IV's writing is great!

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -1851,12 +1858,12 @@
 			]]></tooltip>
-			<story_1><![CDATA[<b>Patient healed:</b>⮀ You're allowed in to see the patients and begin healing. It's excellent practice and everyone you heal, be it from broken bones to small scrapes is extraordinarily thankful for your help.]]></story_1>
-			<story_2><![CDATA[<b>10 Patients healed in a loop:</b>⮀ Your reputation is absolutely sterling, you've basically cleared out the entire healing home and the nurses and healers are more than thankful for your help.]]></story_2>
+			<story_1><![CDATA[<b>Patient healed:</b>⮀ You're allowed in to see the patients and begin healing. You're not too experienced but neither is anyone else here, and your basic magic skills make some of the small things easier even if they can't heal people directly. It's excellent practice and everyone you heal, whether you're curing broken bones or small scrapes, is extraordinarily thankful for ...
```
```suggestion
			<story_1><![CDATA[<b>Patient healed:</b>⮀ You're allowed in to see the patients and begin healing. You're not too experienced but neither is anyone else here, and your basic magic skills make some of the small things easier even if they can't heal people directly. It's excellent practice and everyone you heal, whether you're setting broken bones or bandaging small scrapes, is extraordinarily thankful for your help.]]></story_1>
```

uthbees replied:
I really like this suggestion. This was annoying me earlier, but I couldn't think of anything better. This is great!

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2020,7 +2027,7 @@
             Using your vast knowledge of the dark arts, you can open a rift into the shadow realm to send you straight to Startington.<br>Unfortunately, it doesn't seem like you can fit any supplies through the rift.<br>Unlocked at 300 Dark Magic and 100 Spatiomancy.
             ]]></tooltip>
-            <story_1><![CDATA[<b>Opening the Rift:</b>⮀ Angering the gods just to reach the shadow realm is so much work, and makes you feel a bit bad inside. Thinking back to the lessons from the Witch, you conjure a great deal of destructive and chaotic energy to create a gap between worlds. After a strenuous hour of mentally and physically taxing work, you've done it! A small rift has torn thr ...
+            <story_1><![CDATA[<b>Opening the Rift:</b>⮀ Angering the gods just to reach the shadow realm is so much work, and makes you feel a bit bad inside. Thinking back to your lessons with the witch, you conjure a huge amount of destructive and chaotic energy and begin to fray at the fabric between worlds. After a long and strenuous period of brutally exhausting work, you've done it! A small ...
```
```suggestion
            <story_1><![CDATA[<b>Opening the Rift:</b>⮀ Angering the gods just to reach the shadow realm is so much work, and besides that it makes you feel a bit bad inside. Thinking back to your lessons with the witch, you conjure a huge amount of destructive and chaotic energy, holding it before you, imagining what the villagers would say if they knew that enough power to wipe their village off the map was crackling on their very doorstep. Rather than letting it loose, however, you use your knowledge of Spatiomancy to fold it back on itself over and over again until the angry blot is no larger than your body, and you feel it begin to worry at the fabric between worlds as it squirms to escape your control down the only path you've left open to it. After a long and strenuous period of brutally exhausting focus, you finally feel something give, and you realize you've done it! The dark energy you held so tightly just seems to evaporate, and in its place a small rift has been torn through reality, big enough to fit through and stable enough that it won't shred you to pieces (though you can't say the same for the threadbare bags of traveling supplies you've haggled for so many times). Excited by accomplishing such a feat, you eagerly step across to Startington.]]></story_1>
```

uthbees replied:
This is a bit long - imo the extra details aren't worth the loss of brevity. I did take what I thought were the best changes though.

What I have now could maybe use a little more editing, but then again it's honestly good enough.

dmchurch replied:
Perfect, I love it. I can always revisit the "adding more descriptive text" some other time if I get inspired 😄

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2030,13 +2037,13 @@
-            <story_7><![CDATA[<b>80% Forest explored:</b>⮀ You've managed to figure out where most of the herds in this forest are during your trip through, you've also stumbled across a fair amount of plants that produce mana, you wonder if they have a use.]]></story_7>
-            <story_8><![CDATA[<b>100% Forest explored:</b>⮀ You've done it, you find it impossible to get lost in the forest now, and your sense of wanderlust has faded. You've seen practically all you can in this forest. There were areas barred to you, and you found no way to enter them with a cursory examination.]]></story_8>
+            <story_7><![CDATA[<b>80% Forest explored:</b>⮀ You've managed to figure out the favorite roaming spots of most of the herds in the forest by this point. You've also stumbled across a fair number of funny little green plants that resonate with mana, but in a form that you're unable to absorb. You wonder if they can be used for some other purpose.]]></story_7>
+            <story_8><![CDATA[<b>100% Forest explored:</b>⮀ You've done it, you find it impossible to get lost in the forest now, and your sense of wanderlust has faded. You've seen practically all you can in this forest, although there are still some areas that remain barred to you, with no obvious means of entry revealed by your cursory examination.]]></story_8>
```
```suggestion
            <story_8><![CDATA[<b>100% Forest explored:</b>⮀ You've done it! You find it impossible to get lost in the forest now, and your wanderlust is finally satisfied. You've seen practically all you can in this forest, although there are still some areas that remain barred to you, with no obvious means of entry revealed by your cursory examination.]]></story_8>
```

uthbees replied:
Really great suggestion! This makes it feel much less strange.

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2125,7 +2132,7 @@
             Such simple uses to help out with everyday tasks. Genius, really. It's like pulling teeth getting this knowledge from the Hermit though.<br>Unlocked with both 20% Hermit Knowledge and 50 Magic.
             ]]></tooltip>
-            <story_1><![CDATA[<b>1 Practical magic:</b>⮀ You pester the hermit about magic and how to use it to make things easier on yourself. It's surprisingly handy to be able to move things around without having to be there, rather than smashing pots you can instead lift them or knock them off edges. Why open a lock when you can unlock a latch with Mage Hand, why walk over to the wardrobe if  ...
+            <story_1><![CDATA[<b>1 Practical magic:</b>⮀ You pester the hermit about magic and how to use it to make things easier on yourself. It's surprisingly handy to be able to move things around without having to be there - rather than smashing pots you could instead lift them up and drop them, or knock them off edges. Why pick a lock when you can just unlock the latch with Mage Hand, why w ...
```
```suggestion
            <story_1><![CDATA[<b>1 Practical magic:</b>⮀ You pester the hermit about magic and how to use it to make things easier on yourself. It's surprisingly handy to be able to move things around without having to be there—rather than smashing pots you could instead lift them up and drop them, or knock them off edges. Why pick a lock when you can just unlock the latch with Mage Hand, why walk over to the wardrobe if you can simply fling it open and ransack it from across the room? Magic, but practical. You're a little astonished that not everyone does this.]]></story_1><!-- that's an em dash in there, not a hyphen -->
```

uthbees replied:
Is there any specific reason to change this, especially for this particular story? Ctrl-f picks up 88 instances of hyphens being used like this in game.xml, and I don't really see any need to change that. I guess the unicode symbol is technically more correct, but almost no one uses it, and I don't really see that as a problem. 🤷

dmchurch replied:
Not really no! I hadn't looked at the rest of the file, I was just looking at this in isolation - I make sure to use emdash-with-no-spaces when I'm writing prose, but as you can see I use space-hyphen-space in casual conversation. If the rest of the file is using ` - ` then no reason not to do the same here!

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2219,11 +2226,11 @@
             <story_6><![CDATA[<b>60% Witch knowledge learned:</b>⮀ As you learn more and more about the witch and her varied names, one thing starts to become glaringly obvious. The Witch and The Hermit know one another. If her irritable grumbling about the old fool and his idiotic desires were anything to go by. You get the feeling that their relationship is complex.]]></story_6>
             <story_7><![CDATA[<b>80% Witch knowledge learned:</b>⮀ The witch is struck with a strange sense of melancholy one loop and you have no idea why, but the witch is strangely open with information and you manage to pry out more of the relationship between her and the Hermit. They were lovers once, and to your surprise, she used to be the one whom nature adores. She shuts up quickly after ...
-            <story_8><![CDATA[<b>100% Witch knowledge learned:</b>⮀ Somehow she seems to sense something about you this loop, and she tells you her name, and its the name she gives for every loop after. Trisha. You manage to pry most of her relationship with the hermit from her. They were lovers once, this you knew, but what you didn't was that she was of nobility and he a common vagabond. She fe ...
+            <story_8><![CDATA[<b>100% Witch knowledge learned:</b>⮀ Somehow she seems to sense something about you this loop, and she tells you her name, and it's the name she gives for every loop after. Trisha. You manage to pry most of her relationship with the hermit from her. They were lovers once, this you knew, but what you didn't was that she was of nobility and he a common vagabond. She f ...
```
```suggestion
            <story_8><![CDATA[<b>100% Witch knowledge learned:</b>⮀ Somehow she seems to sense something about you this loop, and she tells you her name, and it's the name she gives for every loop after. Trisha. You manage to pry most of her relationship with the hermit from her. They were lovers once, this you knew, but what you didn't was that she was of nobility and he a common vagabond. She fell for him and he for her; it was a love for the ages, she tells you while drunk on the wine you brought along with you. He wanted to run away with her and she agreed, too lovestruck to question why. He was running from deals he had made and bargains he had forsaken, his glib tongue landing him in trouble of the otherworldly nature. Trisha seems frustrated with her youth's naivete as she tells you how she was made a scapegoat for his failings and forfeits in his deals and bargains. How she was essentially sold to the dark gods as his forfeit. When you ask what she did in revenge, all Trisha does is smile with dark satisfaction, the fire flaring briefly.]]></story_8>
```

uthbees replied:
Nice! Although I think "return" is better than "revenge" here - it seems more evocative of a conversation.

Also, I added the diacritics to "naïveté." Do you know if that could cause problems with fonts or something?

dmchurch replied:
Oh I'm fine with "return"! I was mainly leaning into the "dark evil witch" aspect of things, but given that that's a narration attributed to you and not her, it makes perfect sense to keep it light. And the diacritics are fine! I was debating whether to add them myself but then I thought it might seem too weird 😅

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2560,10 +2567,10 @@
 			<story_2><![CDATA[<b>50 Chronomancy:</b>⮀ The effect is more pronounced now, and you find that your reflexes are keeping up with your increased speed. Unfortunately, though, the mana drain of the loops appears to have kept up as well, so even though you're getting the same amount of work done in less time you can't seem to get more work done in a loop.]]></story_2>
 			<story_3><![CDATA[<b>100 Chronomancy:</b>⮀ You know that speeding up time isn't the only thing time magic can do, but it seems that for a human like you anything fancier is impossible without additional assistance. The runes recommend a group ritual but you're the only chronomancer you know, so instead you start learning what Tachytus has to say about chronomantic potions.]]></story_3>
-            <story_4><![CDATA[<b>1,000 Chronomancy:</b>⮀ How fast can you accelerate the world before you break reality? Who knows, you just want the world to go faster and faster than ever before.]]></story_4>
+            <story_4><![CDATA[<b>1,000 Chronomancy:</b>⮀ You live in a blur, the world flying by as you casually walk. You've learned how to interpret the agonizingly slow sounds that passes for speech among ordinary people, and how to speak the same glacial tongue back at them. You've got places to go, things to do, and never ever enough time to do it all in. You can slow down when you finally l ...
```
Love it 😄

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
             <story_1><![CDATA[<b>1 Mercantilism:</b>⮀ You try your hand at haggling with the God of Trade... and promptly get laughed in your face for your efforts. Thankfully, the God in question is a good sport and helpfully gives you a few tips before sending you on your way.]]></story_1>
-            <story_2><![CDATA[<b>30 Mercantilism:</b>⮀ Over the loops, you learned a few things pretty clearly. For instance, nobody, not even the Gods, likes it when someone tries too hard to haggle. Most people, even the Gods, like a little compliment or a short chat about what's on their mind.]]></story_2>
-            <story_3><![CDATA[<b>100 Mercantilism:</b>⮀ While you're nowhere near the level of talking someone into selling you the clothes off their backs, you have gotten a <i>great</i> feel for what to say when you want a good deal on mana crystals. The loops have never gone this smooth before!]]></story_3>
+            <story_2><![CDATA[<b>30 Mercantilism:</b>⮀ You're pretty good at haggling, if you do say so yourself, but these guys are on a completely different level. They don't need to raise their voice or make wild claims that they'll be left destitute without a better deal, because they're too busy weaving traps in their words without ever breaking their unassailable mask of professionalism. Th ...
```
```suggestion
            <story_2><![CDATA[<b>30 Mercantilism:</b>⮀ You're pretty good at haggling, if you do say so yourself, but these gods are on a completely different level. They don't need to raise their voice or make wild claims that they'll be left destitute without a better deal, because they're too busy weaving traps in their words without ever breaking their unassailable mask of professionalism. There's a lot you can learn here.]]></story_2>
```

uthbees replied:
"Gods" feels less natural to me. How about "people?"

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
-            <story_1><![CDATA[<b>Enchanted armor gained:</b>⮀ Hephestus, God of the Forge, looks at you like you're asking him to forge you a kettle when you ask him to improve your leather armour, but takes the job anyway. "The Gods owe you, after all," he mutters while he works.]]></story_1>
-            <story_2><![CDATA[<b>10 Enchanted armor gained during one loop:</b>⮀ By the time you give Hephestus the final piece of your set, he seems relieved that you won't have any more reason to interrupt him. "Would have turned you away if you didn't get those nice pieces back," he grumbles as he sets to work one more time.]]></story_2>
-            <story_3><![CDATA[<b>25 Enchanted armor gained during one loop:</b>⮀ Hephestus finishes up in record time, but as you reach over to take the finished product, he holds a hand in your way. "Look, the joke has worn out long ago. Don't care if you managed to drag my old hammer back, I'm <i>done</i> doing your dirty work now. Take your armor and leave, or next time I'll throw you out myse ...
+            <story_1><![CDATA[<b>Enchanted armor gained:</b>⮀ The best enchanter in Valhalla has a waiting list equally as impressive. It's not a matter of money, the loop will be finished long before you meet the man. You mention some of this to your good friend the museum curator as you're donating a priceless relic, and he tells you he may have a solution. You're not sure what he did, but an h ...
```
```suggestion
            <story_1><![CDATA[<b>Enchanted armor gained:</b>⮀ The best enchanter in Valhalla has a waiting list equally as impressive. It's not a matter of money; the loop will be finished long before you even meet the man. You mention some of this to your good friend the museum curator as you're donating a priceless relic, and he tells you he may have a solution. You're not sure what he did, but an hour later the god at the top of the list finds you and informs you that he has decided to offer you his spot.]]></story_1>
```

uthbees replied:
Fixed that first sentence, although I used a dash instead of a semicolon. I also changed the line to a list, but I prefer the brevity of the first version to the extra detail of him finding you.

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
-            <story_2><![CDATA[<b>10 Enchanted armor gained during one loop:</b>⮀ By the time you give Hephestus the final piece of your set, he seems relieved that you won't have any more reason to interrupt him. "Would have turned you away if you didn't get those nice pieces back," he grumbles as he sets to work one more time.]]></story_2>
-            <story_3><![CDATA[<b>25 Enchanted armor gained during one loop:</b>⮀ Hephestus finishes up in record time, but as you reach over to take the finished product, he holds a hand in your way. "Look, the joke has worn out long ago. Don't care if you managed to drag my old hammer back, I'm <i>done</i> doing your dirty work now. Take your armor and leave, or next time I'll throw you out myse ...
+            <story_1><![CDATA[<b>Enchanted armor gained:</b>⮀ The best enchanter in Valhalla has a waiting list equally as impressive. It's not a matter of money, the loop will be finished long before you meet the man. You mention some of this to your good friend the museum curator as you're donating a priceless relic, and he tells you he may have a solution. You're not sure what he did, but an h ...
+            <story_2><![CDATA[<b>10 Enchanted armor gained during one loop:</b>⮀ Shoes that step on the air like solid ground, a helmet that deflects projectiles before they even get close, you underestimated the power of enchanted gear. Your every step is filled with augmented vigor and you know that claws and fangs will find no purchase against your reinforced defenses. As a bonus, they're all  ...
```
```suggestion
            <story_2><![CDATA[<b>10 Enchanted armor gained during one loop:</b>⮀ Shoes that step on the air like solid ground, a helmet that deflects projectiles before they even get close, gloves that hold onto your weapons even when you lose your grip—you underestimated the power of enchanted gear. Your every step is filled with augmented vigor and you know that claws and fangs will find no purchase against your reinforced defenses. As a bonus, they're all etched with the elegant patterns and runes you see all over Valhalla, making you look like a great warrior of old.]]></story_2>
```

uthbees replied:
I like it!

dmchurch replied:
rule of threes baybyyyyyyyyyyyyyyyy 😆❤️ (and yeah, 👍 on adjusting the dash style, consistency is best)

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
+            <story_17><![CDATA[<b>Achieved Sage status:</b>⮀ Every single member of the 199 Sages are old enough to have seen countries rise and fall, and the strength they wield reminds you that these are divinity incarnate. You're no slouch though, and you match them spell for spell. Duels at this level are often decided by whoever falters first, and as you create a ludicrously complicated Dark ...
+            <story_18><![CDATA[<b>Achieved Magus status:</b>⮀ The wizard college is too small for a duel against one of the 53 Magi. Your contest takes place in the vast rolling hills outside the city, with ample space to destroy as much land as is necessary to prove your magical might. Your foe does not disappoint, raising pillars of fire taller than the wizard college and then extinguishing it  ...
+            <story_19><![CDATA[<b>Became a member of the Council of the Seven:</b>⮀ Yothik the Serene, grand master of elemental magic, faces you from the hill opposite yours. He's seen you duel your way up to Magus rank during your initiation, and in respect to your skill he's holding nothing back. Two archmages begin casting. The environment around you is incinerated, transmuted, dismantled, we ...
+            <story_20><![CDATA[<b>Became the Chair of the Council of the Seven:</b>⮀ The ultimate authority of the Wizard College is Mimir, God of Knowledge. While his control of magic has never surpassed Odin's, since the dawn of Valhalla Mimir has never once stopped thinking up new ways to utilize magic. The power answers his call like an old friend, gentle and subtle, and at the snap of his fi ...
```
Addressed in actionList.js comments; these want renumbering.

If you want an extra task, however, the format of this XML has always bothered me. If you feel so inclined, feel free to change these (and, potentially, any story you touch, but don't do a global change yet) to the form
```xml
<story num="1" condition="Attempted to attend">Advancement in the Wizard College…</story>
```
and update the related code to support either format.

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
+            The wizard college is the finest school in all the realms. You're lucky to be afforded a chance to get in, although it will cost you a small fortune in tuition.<br>Take their tests and get a grade!<br>Can only have 1 Wizard College action.<br>Costs 500 gold and 10 favors.<br>Gives (all schools of magic skills) * (1 + main stat / 100) * sqrt(1 + times completed / 1000) * (original mana ...
+            ]]></tooltip>
+            <!-- Stories 1-11 have been replaced. Don't use those numbers or your stories may not show in the action log properly on old saves. -->
+            <story_12><![CDATA[<b>Attempted to attend:</b>⮀ Advancement in the Wizard College is determined by wizard duels. To rise to a title higher than Initiate, you must challenge someone at that rank and duel them for it. Every trick you can perform that they can't wins you a point, and vice versa. It looks like you'll need skills in all kinds of magic to succeed here.]]></story_12>
```
Add something here about "at least they let you audit their classes as long as you paid the entrance fee" to reflect that WC is now required for Spatio and Resto training, along with an explanation of why the favors are needed?

uthbees replied:
I like that! I wrote that Initiates can go to classes but not really do anything else, since that fits in really nicely with the Student story (even if the rank titles don't really match up :P).

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
-            <story_2><![CDATA[<b>50 Restoration:</b>⮀ Not only do you now know where to find all the tools the infirmary has, you also know how to use them without making things worse! The additional lessons in simple healing cantrips are moving from "heal lesser bruise" to something a bit more potent, too!]]></story_2>
-            <story_3><![CDATA[<b>200 Restoration:</b>⮀ You're really starting to appreciate how the curriculum started with the "mundane" ways of healing the infirm as you fight to keep your current patient from dying after a first-year <i>horrendously</i> messed up a healing spell they only kinda understood. If nothing else, it gave you the iron stomach to stay focused while removing the mangled ...
-            <story_4><![CDATA[<b>500 Restoration:</b>⮀ Figuring out a passive healing aura had taken a while, but you managed and now people near you can fall flat on their face without getting so much as a scratch. While you're no longer allowed on the main infirmary floor (because the other students couldn't get in their practice time with you around,) you're now assigned to the hairier cases i ...
+            <story_1><![CDATA[<b>Practiced Restoration:</b>⮀ You gather in a room full of cloaked wizards to discuss the nature of life energy. It's quicker, far quicker, than the healing you're used to. Flesh and blood sing in response to it, cleansing impurities and fixing maladies in the blink of an eye. The key is control, to master Restoration you must be able to take life itself in your han ...
```
```suggestion
            <story_1><![CDATA[<b>Practiced Restoration:</b>⮀ You gather in a room full of cloaked wizards to discuss the nature of life energy. It's quicker, far quicker, than the healing you're used to. Flesh and blood sing in response to it, cleansing impurities and fixing maladies in the blink of an eye. The key is control; to master Restoration you must be able to take life itself in your hands.]]></story_1>
```

uthbees replied:
Nice - done.

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
-            <story_3><![CDATA[<b>200 Restoration:</b>⮀ You're really starting to appreciate how the curriculum started with the "mundane" ways of healing the infirm as you fight to keep your current patient from dying after a first-year <i>horrendously</i> messed up a healing spell they only kinda understood. If nothing else, it gave you the iron stomach to stay focused while removing the mangled ...
-            <story_4><![CDATA[<b>500 Restoration:</b>⮀ Figuring out a passive healing aura had taken a while, but you managed and now people near you can fall flat on their face without getting so much as a scratch. While you're no longer allowed on the main infirmary floor (because the other students couldn't get in their practice time with you around,) you're now assigned to the hairier cases i ...
+            <story_1><![CDATA[<b>Practiced Restoration:</b>⮀ You gather in a room full of cloaked wizards to discuss the nature of life energy. It's quicker, far quicker, than the healing you're used to. Flesh and blood sing in response to it, cleansing impurities and fixing maladies in the blink of an eye. The key is control, to master Restoration you must be able to take life itself in your han ...
+            <story_2><![CDATA[<b>50 Restoration:</b>⮀ The teachers must be confident in their abilities. You've reached the advanced classes, and now you get to test on real, injured people. Volunteers, willingly wounded every which way, just to give your class practice. You feel the gaze of the professor upon you as you heal a nearly-amputated arm. He'd step in before it's too late, of course, b ...
```
```suggestion
            <story_2><![CDATA[<b>50 Restoration:</b>⮀ The teachers must be confident in their abilities. You've reached the advanced classes, and now you get to test on real, injured people. Volunteers willingly allow themselves to be wounded every which way, just to give your class practice. You feel the gaze of the professor upon you as you heal a nearly-amputated arm. He'd step in before it's too late, of course, but exactly how close to death would that be?]]></story_2>
```

uthbees replied:
This still feels a little weird to me after the edit, but I can't come up with anything better off the top of my head. Good enough.

dmchurch replied:
Yeah, it's still a little clunky, but eh 🤷 Good enough indeed!

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
-            <story_4><![CDATA[<b>500 Restoration:</b>⮀ Figuring out a passive healing aura had taken a while, but you managed and now people near you can fall flat on their face without getting so much as a scratch. While you're no longer allowed on the main infirmary floor (because the other students couldn't get in their practice time with you around,) you're now assigned to the hairier cases i ...
+            <story_1><![CDATA[<b>Practiced Restoration:</b>⮀ You gather in a room full of cloaked wizards to discuss the nature of life energy. It's quicker, far quicker, than the healing you're used to. Flesh and blood sing in response to it, cleansing impurities and fixing maladies in the blink of an eye. The key is control, to master Restoration you must be able to take life itself in your han ...
+            <story_2><![CDATA[<b>50 Restoration:</b>⮀ The teachers must be confident in their abilities. You've reached the advanced classes, and now you get to test on real, injured people. Volunteers, willingly wounded every which way, just to give your class practice. You feel the gaze of the professor upon you as you heal a nearly-amputated arm. He'd step in before it's too late, of course, b ...
+            <story_3><![CDATA[<b>200 Restoration:</b>⮀ Now you're the one cleaning up after new student's mistakes, although the professors are still around to step in if you get in over your head. You mentally curse the college's lax safety policies as you fight to keep your current patient from dying after a first-year <i>horrendously</i> messed up a healing spell they only kinda understood. On ...
```
```suggestion
            <story_3><![CDATA[<b>200 Restoration:</b>⮀ Now you're the one cleaning up after new students' mistakes, although the professors are still around to step in if you find yourself in over your head. You mentally curse the college's lax safety policies as you fight to keep your current patient from dying after a first-year <i>horrendously</i> messed up a healing spell they only kinda understood. On the other hand, if the policies weren't so lax, you'd never have gained the iron stomach that's allowing you to stay focused while removing swaths of your patient's mangled flesh.]]></story_3>
```

uthbees replied:
Great suggestions, thanks! I wrote the beginning and ending of this while merging IV's and Vera's restoration stories, so this is the first time these lines have been edited.

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2819,57 +2829,56 @@
+            <story_1><![CDATA[<b>Practiced Restoration:</b>⮀ You gather in a room full of cloaked wizards to discuss the nature of life energy. It's quicker, far quicker, than the healing you're used to. Flesh and blood sing in response to it, cleansing impurities and fixing maladies in the blink of an eye. The key is control, to master Restoration you must be able to take life itself in your han ...
+            <story_2><![CDATA[<b>50 Restoration:</b>⮀ The teachers must be confident in their abilities. You've reached the advanced classes, and now you get to test on real, injured people. Volunteers, willingly wounded every which way, just to give your class practice. You feel the gaze of the professor upon you as you heal a nearly-amputated arm. He'd step in before it's too late, of course, b ...
+            <story_3><![CDATA[<b>200 Restoration:</b>⮀ Now you're the one cleaning up after new student's mistakes, although the professors are still around to step in if you get in over your head. You mentally curse the college's lax safety policies as you fight to keep your current patient from dying after a first-year <i>horrendously</i> messed up a healing spell they only kinda understood. On ...
+            <story_4><![CDATA[<b>500 Restoration:</b>⮀ You see now why the professors were so bold. You're on the same level as the professors now, and at this point, if you saw someone breathing their last dying gasp, you could have them back up on their feet in mere moments. When you fight alongside allies, you can seal up cuts as they're happening, undoing damage that's still being dealt. You' ...
```
```suggestion
            <story_4><![CDATA[<b>500 Restoration:</b>⮀ You see now why the professors were so bold. You're on the same level as they are now, and at this point, if you saw someone breathing their last dying gasp, you could have them back up on their feet in mere moments. When you fight alongside allies, you can seal up cuts as they're happening, undoing damage that's still being dealt. You've heard wizards say that without Restoration, the gods would never have been able to conquer Valhalla back at the dawn of history, and at this point you believe it.]]></story_4>
```

uthbees replied:
Also great!

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2905,27 +2914,30 @@
+            <story_1><![CDATA[<b>Visited the Oracle:</b>⮀ The oracle explains that every person has a hamingja, a guardian spirit that decides their luck, and that if you play a few games of chance she can read its energies and predict your fortune. A couple rounds of pretend gambling later and she clucks her tongue in disappointment. Your hamingja is apparently slacking off, because she's barely ...
+            <story_2><![CDATA[<b>100 Luck talent:</b>⮀ It's a subtle trend, barely noticeable, but you're rolling higher than you used to.  If you flip 10 coins in a row you'll get 6 heads more often than 6 tails. You must have finally gotten your hamingja to pay attention to you, though if you weren't keeping a mental tally of your results you'd never have noticed.]]></story_2>
+            <story_3><![CDATA[<b>1,000 Luck talent:</b>⮀ You spot a coin lying on the side of the road as you walk by. That never used to happen back when you were a simple courier. It's hard to imagine how luck could've caused a coin to land here, but you suppose you might've just walked by it if the sunlight didn't happen to glint off of it as you pass by. You send a flash of gratitude to your  ...
+            <story_4><![CDATA[<b>10,000 Luck talent:</b>⮀ If you really want to, you can flip ten coins and call it right eight out of ten times every time. This makes you an absolute monster at the casino, enough to easily get you accused of cheating. But it'd be bad if you got kicked out, so your luck is always just believable enough to keep people from complaining. Cheating is for chumps anywa ...
```
```suggestion
            <story_4><![CDATA[<b>10,000 Luck talent:</b>⮀ If you really want to, you can flip ten coins and call it right eight out of ten times every time. This makes you an absolute monster at the casino, enough to easily get you accused of cheating. But it'd be bad if you got kicked out, so your luck is always just believable enough to keep people from complaining. Cheating is for chumps, anyways; you don't need an ace up your sleeve if you know you'll always draw one when you really need it.]]></story_4>
```

uthbees replied:
Nice!

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2905,27 +2914,30 @@
-            <story_2><![CDATA[<b>100 Luck talent:</b>⮀ The Oracle double-takes when she tries and read your fate. "How strange... For mortals, the past is only ever a single line, but your past branches like the roots of a tree," she explains when you ask.]]></story_2>
-            <story_3><![CDATA[<b>1,000 Luck talent:</b>⮀ The Oracle gasps dramatically as she performs her reading. "Fortune favours you in strange ways, stranger! You will face impossible challenges in the hours ahead, and yet... you shall pass them without a single scratch!" Something about the way she says it feels less like a prediction, and more like she's speaking it into existence.]]></sto ...
-            <story_4><![CDATA[<b>10,000 Luck talent:</b>⮀ You barely get settled down for your Reading when a bony hand rests on the Oracle's shoulder. "Sorry, dearie, but would you mind if we handled this one?" asks the trio of definitely-not-there before Goddesses of Fate. The Oracle visibly sags but steps to the side. The Goddesses look at you. "Will you do it, then? Will you climb the Tower i ...
+            <!-- Note: Hamingja are a real thing in norse mythology!  It's perhaps the most blatant non-pop-culture
+             Norse-ism so far, but I like the way it frames the stories here. -Inferno Vulpix -->
```
I do too. 💖

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -2999,15 +3011,15 @@
-            <story_8><![CDATA[<b>100% Explored:</b>⮀ No matter how hard the town tried to confuse and mislead you, you're now pretty sure that you've got the entire layout memorized. It's still not a walk in the park, but at least you can now reliably make your way from your starting position to the stubby tower along the path out of town.]]></story_8>
-            <story_9><![CDATA[<b>Meandered with 100 Imbue Mind:</b>⮀ If you focus, you can still feel the gloom pull on your thoughts, but by this point it feels like a gentle breeze getting upset you won't dance like the leaves it carries.]]></story_9>
+            <story_1><![CDATA[<b>1% Explored:</b>⮀ This place is disgusting. Filthy rats scurry under rubble and the derelict roads are frequently marred by bloodstains. This place would set you on edge even without the headache-inducing mental effect that even now makes your mind swim. It takes all you have to keep walking forwards, onwards into this horrific landscape.]]></story_1>
+            <story_2><![CDATA[<b>2% Explored:</b>⮀ Your shambling walk comes to a halt as you reach a plaza with a well in the middle. You recognize places like this from Beginnersville, people loved to come to the well and gossip as they drew water for their houses. The plaza here is empty.]]></story_2>
```
```suggestion
            <story_2><![CDATA[<b>2% Explored:</b>⮀ Your shambling walk comes to a halt as you reach a plaza with a well in the middle. You recognize places like this from Beginnersville: people loved to come to the well and gossip as they drew water for their houses. The plaza here is empty.]]></story_2>
```

uthbees replied:
Done.

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -3017,17 +3029,17 @@
+            <story_1><![CDATA[<b>Mana Well drained:</b>⮀ There must have been a ley line beneath the town, because deep within the wells here you sense mana. It's fleeting though, fading rapidly with every moment you stand here. What's causing this? Where is the mana all going? With no time to waste, you begin hauling up water so you can start extracting mana before it's all gone.]]></story_1>
+            <story_2><![CDATA[<b>10 Mana Wells with mana found:</b>⮀ You can tell what's happening to all the mana now. It's all being siphoned into that great big spire in the middle of town, the same one that's drawing power from all those pylons. The entire town is being drained, crumbling to pieces as the evil structure takes and takes and takes. You don't know what will happen when there's n ...
+            <story_3><![CDATA[<b>15 Mana Wells with mana found:</b>⮀ There are ruins beneath the rubble, and wells that have long since been dismantled and forgotten. Under a house's foundation or beneath the stones of an empty plaza, accessible with the right tools and knowledge. But yet, even here the mana is quickly drained away. You had hoped to see how much mana these wells had before the dr ...
+            <story_4><![CDATA[<b>Drawn from an empty well:</b>⮀ Nope, bone dry. There's still water in here, murky and probably very unhealthy water, but not a speck of mana. You toss the bucket back down the well and sigh. At least now you know: if you ever take this long getting here, just don't bother.]]></story_4>
```
```suggestion
            <story_4><![CDATA[<b>Drawn from an empty well:</b>⮀ Nope, bone dry. There's still water in here, murky and probably very unhealthy water, but not a speck of mana. You toss the bucket back down the well and sigh. At least now you know: if you can't get here any faster than you did just now, just don't bother.]]></story_4>
```

uthbees replied:
I'm sort of of two minds about this one, so I'll take the version you prefer.

dmchurch commented on lang/en-EN/game.xml:
```diff
@@ -3100,7 +3114,7 @@
             Costs 500 gold.<br>You only need one set of supplies.<br>Unlocked with 75% town meandered.
             ]]></tooltip>
-            <story_1><![CDATA[<b>Supplies bought:</b>⮀ Try as you might, the merchant seems completely deaf at your every attempt at haggling. While you aren't happy to part with half a thousand gold, you do so anyway and take your bundle of supplies. In retrospect, the fact that the merchant had nothing else for sale suggests business isn't going all that smoothly.]]></story_1>
+            <story_1><![CDATA[<b>Supplies bought:</b>⮀ The man who sells you these supplies, you can tell that he doesn't have much more than what he sold you. You try not to think about it, but you can't forget the bone-deep weariness in his movements or the resolute expression on his face. He knows exactly what he's doing, and you understand why your best efforts couldn't drive the price down e ...
```
```suggestion
            <story_1><![CDATA[<b>Supplies bought:</b>⮀ The man who sells you these supplies, you can tell that he doesn't have much more than what he sold you. You try not to think about it, but you can't forget the bone-deep weariness in his movements or the resolute expression on his face. He knows exactly what he's doing, and you understand why your best efforts couldn't drive the price down even a single gold. But you have places to be, and you very dearly want to get out of here. You buy the supplies, cursing the town's name under your breath.]]></story_1>
```

uthbees replied:
Done.

dmchurch commented on views/main.view.js:
```diff
@@ -1026,33 +1026,43 @@ class View {
-                            lastInBranch = false;
-                            storiesUnlocked++;
-                            if (action.visible() && action.unlocked() && completedActions.includes(action.varName)) {
-                                actionLog.addActionStory(action, i, init);
+
+                    const rawStoriesDataForAction = _txtsObj(`actions>${name}`, "fallback")[0].children;
```
The `Action` class, since it extends `Localizable`, has a `.txtsObj` property which will give you the proper XML object for that action's localizations. You can use that instead of calling `_txtsObj`, but I think I'd prefer if you take it a step further; add a method to `Action` called, say, `getStoryTexts()`, that retrieves an array consisting of all applicable stories, and which can take care of all the `.split("⮀")` nonsense on its own. (should be a method, rather than a property accessor, since one of the changes I'll eventually be making is that a single `Action` can be valid/present in all zones, but may have different stories depending on which it's performed in.)

dmchurch commented on views/main.view.js:
```diff
@@ -1328,23 +1338,29 @@ class View {
-                } else {
-                    storyTooltipText += `${storyText[0]} ???`;
-                    lastInBranch = true;
+            const name = action.name.toLowerCase().replace(/ /gu, "_");
+
+            const rawStoriesDataForAction = _txtsObj(`actions>${name}`, "fallback")[0].children;
```
this also wants to use `.txtsObj` or `.getStoryTexts()`

uthbees committed:
Misc story tweaks/fixes

dmchurch reviewed (changes requested):
This looks great, and the writing is top-notch! One thing you'll need to add is some logic to the ActionLog on what to do if it encounters an action story that's no longer defined in the language file. Feel free to add some creative text like "You remember only that something happened here, but not what it was" or the like, it can go with all the rest of the AL-specific strings.

I've included a number of editorial suggestions to the writing as well as a little bit of my own; even more than with the notes on the code itself, please take these as suggestions only and feel free to edit, expand, or even remove them entirely! These will be going into the repo with your username on the blame line, after all, and if I feel strongly about the edits I can always make them in a followup commit of my own.

uthbees commented:
Oh wow, that's a lot of suggestions! I'll definitely start going through these when I get a chance. It might take me a while to get through them all though :)

dmchurch commented:
No rush! I got very excited so I spent all day making sure to give it a very careful review ♥

uthbees committed:
Handle missing action stories more gracefully in the log

uthbees committed:
Misc Valhalla story fixes

uthbees committed:
Edit pyromancy stories

uthbees committed:
Misc story improvements

Thanks to dmchurch for the suggestions!

uthbees commented:
That should be all the content suggestions addressed. Everything else is just refactoring for maintainability and such (from the initial review, anyway). I can definitely make at least a couple of the other changes before we merge, but that's not as high of a priority - the user-facing stuff is back to you.

dmchurch commented:
This is all fantastic! I'm 100% happy with all the stories. Do you want me to knock out the technical stuff? I'm happy to do it if it doesn't excite you, but I know I at least sometimes enjoy getting invited to do a refactor on someone else's project, so I wanted to leave the option open to you ❤️

uthbees commented:
> Do you want me to knock out the technical stuff?

Either way is fine by me, although merging would probably happen a lot sooner if you did it. :) I definitely don't mind doing it myself, but contributing to this project is pretty much my last priority, so it could be a while before I get around to everything.

uthbees committed:
Merge remote-tracking branch 'upstream/dmchurch' into dmchurch

dmchurch commented:
Sure, no problem! I'll go ahead and take care of that stuff, then - my wife has been bugging me about "hey when do we get all the new stories?" 😂

dmchurch committed:
Renaming storyReqs to storyFlags and unlockStory to setStoryFlag

dmchurch committed:
Adding storyVar functionality and using it to simplify WC code

dmchurch committed:
Centralized the action story parsing code; adding new syntax for action stories in game.xml

Also, noting that WC enrollment is required for taking classes in their tooltips.
